parsers/json/json.cc

```diff
@@ -0,0 +1,108 @@
+#include "json.h"
+
+namespace
+{
+  using namespace trieste;
+  using namespace trieste::json;
+
+  std::size_t invalid_tokens(
+    Node n, std::initializer_list<Token> tokens, const std::string& message)
+  {
```

Comment:
Given how you use this, would it make sense for it to be 
```C++
std::initializer_list<std::pair<Token,const std::string>> token_messages
```
Then there would only be a single pass, if you are trying to check for several tokens.

Reply:
yeah that's a great suggestion

Reply:
Changed.